masq/src/commands/financials_command.rs

```diff
@@ -402,319 +377,183 @@ impl FinancialsCommand {
-            h_and_w[0].0,
-            h_and_w[1].0,
-            h_and_w[2].0,
-            h_and_w[3].0,
-            h_and_w[4].0,
-            ordinal_num_width = h_and_w[0].1,
-            wallet_width = WALLET_ADDRESS_LENGTH,
-            age_width = h_and_w[2].1,
-            balance_width = h_and_w[3].1,
-            hash_width = h_and_w[4].1
+    fn convert_masq_from_gwei_and_dress_well<T>(gwei: T) -> String
+    where
+        T: Display + PartialEq + From<u32> + Div<Output = T> + PartialOrd<T>,
+        <T as Div<T>>::Output: PartialEq<T> + PartialOrd<T> + Display,
+    {
+        let (affected_part, is_positive) = match Self::preformatting_math_analysis(gwei) {
+            Left(result) => return result,
+            Right(tuple) => tuple,
+        };
+        let minimal_padding = if is_positive { 3 } else { 4 };
+        let result = format!("{:0count$}", affected_part, count = minimal_padding);
+        format!(
+            "{}.{}",
+            &result[..result.len() - 2].separate_with_commas(),
+            &result[result.len() - 2..]
         )
     }
 
-    fn zip_them<'a>(
-        words: &'a [&'a str],
-        optimized_widths: &'a [usize],
-    ) -> Vec<(&'a str, &'a usize)> {
-        words.iter().copied().zip(optimized_widths.iter()).collect()
+    fn preformatting_math_analysis<T>(gwei: T) -> Either<String, (T, bool)>
+    where
+        T: Display + PartialEq + From<u32> + Div<Output = T> + PartialOrd<T>,
+        <T as Div<T>>::Output: PartialEq<T> + PartialOrd<T> + Display,
+    {
+        if gwei == T::from(0) {
+            return Left("0".to_string());
+        }
+        let is_positive = gwei >= T::from(0);
+        let affected_part = gwei / T::from(10_000_000);
+        if affected_part == T::from(0) {
+            return Left(
+                if is_positive {
+                    "< 0.01"
+                } else {
+                    "-0.01 < x < 0"
+                }
+                .to_string(),
+            );
+        }
+        Right((affected_part, is_positive))
     }
 
-    fn render_single_payable(
-        stdout: &mut dyn Write,
-        preprocessed_values: &[String],
-        optimal_widths: &[usize],
-        ordinal_num: usize,
-    ) {
+    fn financial_status_totals_title(stdout: &mut dyn Write, gwei: bool) {
         short_writeln!(
             stdout,
-            "{:<ordinal_num_width$}   {:wallet_width$}   {:<age_width$}   {:<balance_width$}   {:<hash_width$}",
-            ordinal_num,
-            preprocessed_values[0],
-            preprocessed_values[1],
-            preprocessed_values[2],
-            preprocessed_values[3],
-            ordinal_num_width = optimal_widths[0],
-            wallet_width = WALLET_ADDRESS_LENGTH,
-            age_width = optimal_widths[2],
-            balance_width = optimal_widths[3],
-            hash_width = optimal_widths[4]
-        )
+            "\nFinancial status totals in {}\n",
+            &Self::gwei_or_masq_balance(gwei)[9..13]
```

Comment:
Couple things here.
First, as you pointed out, `is_gwei` would be a better name than `gwei`.
Second, it would be clearer to have the function that produced either "Gwei" or "MASQ" just return one of those strings, and have `gwei_or_masq_balance(bool)` call it. Then you could call that function here, and its operation would be obvious.